### PR TITLE
Focus the window before doing the Shift+Insert

### PR DESCRIPTION
**Pull request by chmduquesne**
This is the only way I found to make rofimoji work with firefox3: using the clipboard and focusing the window beforehand.

I also have this snippet, that I ended up not using:

    def is_firefox(windowid) -> bool:
        xprop = Popen(args=['xprop', '-id', windowid, 'WM_CLASS'], stdout=PIPE)
        output = xprop.communicate()[0].decode("utf-8")[:-1]
        wmclass = map(lambda x: x.strip(), output.split('=')[1].replace('"', '').split(','))
        return 'firefox' in wmclass

### CODE REVIEW

**Comment by fdw** [comment]
Intersting. Do you have focus follows mouse active?

**Comment by chmduquesne** [comment]
I am using it in i3, so yes

**Comment by chmduquesne** [comment]
By the way, this is using the "command chaining" feature of xdotool (see COMMAND CHAINING in the manpage). It could be done in 2 separate commands:

    xdotool windowfocus <active_window> key --clearmodifiers 'Shift+Insert'

is equivalent to

    xdotool windowfocus <active_window>
    xdotool key --clearmodifiers 'Shift+Insert'

Note that by dropping the `--window <active_window>`, we are moving from XSendEvent, which firefox ignores, to XTEST, which it accepts (see SENDEVENT NOTES in the manpage). Unfortunately firefox 3 does not let xdotool type emojis with the XTEST events (I tried), but it accepts normal text.

Long story short, the only solution that I found for for firefox is therefore to use the clipboard and send a Shift+Insert as an XTEST.

**Comment by fdw** [comment]
Thanks! 😃
